graphbolt/src/fused_csc_sampling_graph.cc

```diff
@@ -484,6 +495,309 @@ auto GetTemporalPickFn(
+                  num_picked_neighbors_data_ptr
+                      [etype_id_to_num_picked_offset[i]] = 0;
+                }
+                substract_offset =
+                    substract_offset.cumsum(0, indptr_.scalar_type());
+                subgraph_indptr_substract = ops::ExpandIndptr(
+                    num_picked_offset_tensor, indptr_.scalar_type(),
+                    substract_offset);
+              }
+
+              // Step 2. Calculate prefix sum to get total length and offsets of
+              // each node. It's also the indptr of the generated subgraph.
+              subgraph_indptr = num_picked_neighbors_per_node.cumsum(
+                  0, indptr_.scalar_type());
+              auto subgraph_indptr_data_ptr =
+                  subgraph_indptr.data_ptr<indptr_t>();
+
+              // When doing non-temporal hetero sampling, we generate an
+              // edge_offsets tensor.
+              if (type_per_edge_.has_value() && fanouts.size() > 1 &&
+                  with_seed_offsets) {
+                edge_offsets = torch::empty({num_etypes + 1}, indptr_options);
+                AT_DISPATCH_INTEGRAL_TYPES(
+                    edge_offsets.value().scalar_type(), "CalculateEdgeOffsets",
+                    ([&] {
+                      auto edge_offsets_data_ptr =
+                          edge_offsets.value().data_ptr<scalar_t>();
+                      edge_offsets_data_ptr[0] = 0;
+                      for (auto i = 0; i < num_etypes; ++i) {
+                        edge_offsets_data_ptr[i + 1] = subgraph_indptr_data_ptr
+                            [etype_id_to_num_picked_offset[i + 1] - 1];
+                      }
+                    }));
+              }
+
+              // Step 3. Allocate the tensor for picked neighbors.
+              const auto total_length =
+                  subgraph_indptr.data_ptr<indptr_t>()[num_rows - 1];
+              picked_eids = torch::empty({total_length}, indptr_options);
+              subgraph_indices =
+                  torch::empty({total_length}, indices_.options());
+              if ((fanouts.size() == 1 || !with_seed_offsets) &&
+                  type_per_edge_.has_value()) {
+                subgraph_type_per_edge = torch::empty(
+                    {total_length}, type_per_edge_.value().options());
+              }
+
+              auto picked_eids_data_ptr = picked_eids.data_ptr<indptr_t>();
+              torch::parallel_for(
+                  0, num_seeds, grain_size, [&](int64_t begin, int64_t end) {
+                    for (int64_t i = begin; i < end; ++i) {
+                      const auto nid = seeds_data_ptr[i];
+                      const auto offset = indptr_data[nid];
+                      const auto num_neighbors = indptr_data[nid + 1] - offset;
+                      auto picked_number = 0;
+                      const auto seed_type_id =
+                          (with_seed_offsets && fanouts.size() > 1)
```

Comment:
let's refactor `with_seed_offsets && fanouts.size() > 1` into a variable as it is used commonly.